bin/fuel-core/src/cli/run.rs

```diff
@@ -250,11 +256,13 @@ impl Command {
         let (chain_conf, state_config) = match genesis_config.as_deref() {
             None => (ChainConfig::local_testnet(), StateConfig::local_testnet()),
             Some(path) => {
-                let chain_conf = ChainConfig::load_from_directory(path)?;
-                let state_config = StateConfig::load_from_directory(path)?;
+                let metadata = SnapshotMetadata::read(path)?;
+                let chain_conf = ChainConfig::from_snapshot(&metadata)?;
```

Comment:
I think the name of the function would make more sense as `from_snapshot_metadata`.

Reply:
I'm a b it hesitant about `from_snapshot_metadata` since it makes it sound as if the config is loaded from the metadata file instead of the snapshot.
Unless that gives you second thoughts as well, I'm ready to go with `from_snapshot_metadata`

Reply:
Maybe I'm just misreading what's happening. It looks like you're loading the `SnapshotMetadata` and passing it to the `from_snapshot` method. i.e. you're passing the metadata, not the snapshot to load from. 

Just seems like a mismatch of names. Are you really loading from the snapshot, not the metadata?